apiserver/uniter/uniter.go

```diff
@@ -430,7 +430,26 @@ func (u *UniterAPIV3) GetPrincipal(args params.Entities) (params.StringBoolResul
 			if err == nil {
 				principal, ok := unit.PrincipalName()
 				if principal != "" {
-					result.Results[i].Result = names.NewUnitTag(principal).String()
+					result.Results[i].Result = principal
+				} else {
+					// Note unit.PrincipalName() doesn't return a name for subordinates! It
```

Comment:
unit.PrincipalName *only* returns a non-empty value for subordinates
can you please explain the scenario you tested which required this change? clearly I'm still missing something from out email discussion